float64 givenX
float64 givenY
float64 givenTheta
---